### ISSUE TITLE
Deploy with CodeDeploy

### ISSUE THREAD

**Issue by markmac99** [markmac99/UKmon-shared#169]
This would make it easier to share the development tasks, since the CodeDeploy process can be controlled via AWS IAM keys. The ansible-based process is very robust but hard to share.

**Comment by markmac99** [comment]
Unfortunately CodeDeploy has some severe limitations, and isn't suitable. There are two probems
1) it totally destroys the existing folders before redeploying. 
2) It doesn't suport deploying prod and dev from the same repo. Which is a ridiculous restriction!

Specifically, codedeploy reads an appspec file from the root of the code repository that contains all the target paths to which code will be deployed. There can be only one appspec file, so if you want to deploy prod and dev to different locations, you need to change the file. HOWEVER, if you change the file, codedeploy's first action is to delete the previous target folders. So if you deploy to ~/dev, then change the target to ~/prod, codedeploy will delete ~/dev. 

Having read up on this, AWS's position is that a) you should be using a different server for dev and prod, and b) if you want to deploy different files or to a different location, you should put the code in a differnt repo. The first is just about plausible, though it has unacceptable cost implications for SMEs. The 2nd is ridiculous, and counter to all best-practice which says that prod and dev code should be branches in the same repo.  AWS have been asked to support deployment from branches but apparently refused. 

So, TLDR, code-deploy is not useable. Back to Ansible it is.
